### PR TITLE
Fix serialization of arrays

### LINKED ISSUE #1505: LogsAggregateRequest response returns null values for log timeseries requests
**Describe the bug**
JSON marshaling of aggregate log timeseries response is broken and always returns null values for the timeseries.

**To Reproduce**

1. Execute [this example](https://github.com/DataDog/datadog-api-client-go/blob/master/examples/v2/logs/AggregateLogs_2527007002.go).
2. Notice null values.

**Expected behavior**
Values are returned for each timeseries that is produced by the query.

**Screenshots**
image

**Environment and Versions (please complete the following information):**
A clear and precise description of your setup:

datadog-api-client-go v.1.14.0
golang 1.16.5

**Additional context**

### PR DESCRIPTION
**Pull request by therve**
This was broken in #1376.

Fixes #1505.

### CODE REVIEW

File: .generator/src/generator/templates/model_simple.j2 [hunk 1]
```diff
@@ -249,33 +249,14 @@ func (o *{{ name }}) Unset{{ propertyName }}() {
...
-	}
-	{{/isArray}}
-	{{/isMap}}
-	{{#isArray}}
 	if o.UnparsedObject != nil {
 		return json.Marshal(o.UnparsedObject)
 	}
+{%- if model.type == "array" %}
 	for i, item := range o.Items {
 		toSerialize[i] = item
```
Comment (NouemanKHAL) [inline .generator/src/generator/templates/model_simple.j2:257]: Could `Items` be an array of pointers? (we might need to copy `item` before assigning it `toSerialize`)

Comment (therve) [reply .generator/src/generator/templates/model_simple.j2:257]: What would we need to? We're not modifying them, do we?

Comment (NouemanKHAL) [reply .generator/src/generator/templates/model_simple.j2:257]: Yes but if `item` is a pointer, then after the loop ends, all `toSerialize` items will point to the last item.

Comment (therve) [reply .generator/src/generator/templates/model_simple.j2:257]: I was not able to get a pointer in there, so I guess we should be fine.

**Review by NouemanKHAL — APPROVED** [review]

**Pull request merged**